diff -urN orig/afnix-src-4.0.0/cnf/bin/afnix-vcomp afnix-src-4.0.0/cnf/bin/afnix-vcomp
--- orig/afnix-src-4.0.0/cnf/bin/afnix-vcomp	2024-12-24 09:36:45.000000000 -0500
+++ afnix-src-4.0.0/cnf/bin/afnix-vcomp	2026-07-22 19:52:11.117221113 -0400
@@ -120,6 +120,8 @@
        12*) ccvers=12 ;;
        13*) ccvers=13 ;;
        14*) ccvers=14 ;;
+       15*) ccvers=15 ;;
+       16*) ccvers=16 ;;
 	esac
     fi
     # check for g++ version
@@ -134,7 +136,10 @@
        10*) ccvers=10 ;;
        11*) ccvers=11 ;;
        12*) ccvers=12 ;;
+       13*) ccvers=13 ;;
        14*) ccvers=14 ;;
+       15*) ccvers=15 ;;
+       16*) ccvers=16 ;;
 	esac
     fi
     # check for clang version
diff -urN orig/afnix-src-4.0.0/cnf/mak/afnix-gc15.mak afnix-src-4.0.0/cnf/mak/afnix-gc15.mak
--- orig/afnix-src-4.0.0/cnf/mak/afnix-gc15.mak	1969-12-31 19:00:00.000000000 -0500
+++ afnix-src-4.0.0/cnf/mak/afnix-gc15.mak	2026-07-22 19:54:52.706649872 -0400
@@ -0,0 +1,178 @@
+# ----------------------------------------------------------------------------
+# - afnix-gcc15                                                              -
+# - afnix compiler configuration - gcc configuration                         -
+# ----------------------------------------------------------------------------
+# - This program is  free software;  you can  redistribute it and/or  modify -
+# - it provided that this copyright notice is kept intact.                   -
+# -                                                                          -
+# - This  program  is  distributed in the hope  that it  will be useful, but -
+# - without  any   warranty;  without  even   the   implied    warranty   of -
+# - merchantability  or fitness for a particular purpose. In not event shall -
+# - the copyright holder be  liable for  any direct, indirect, incidental or -
+# - special damages arising in any way out of the use of this software.      -
+# ----------------------------------------------------------------------------
+# - author (c) 1999-2025 amaury   darsch                                     -
+# - author (c) 2011-2013 pino     toscano                      hurd platform -
+# - author (c) 2016-2016 martin   michlmayr                           GCC 05 -
+# - author (c) 2016-2017 amaury   darsch                              GCC 06 -
+# - author (c) 2017-2017 nobuhiro iwamatsu                            GCC 07 -
+# - author (c) 2018-2018 nobuhiro iwamatsu                            GCC 08 -
+# - author (c) 2022-2022 nobuhiro iwamatsu                            GCC 12 -
+# - author (c) 2019-2023 amaury   darsch                     GCC 09/10/11/12 -
+# - author (c) 2022-2023 amaury   darsch                              GCC 13 -
+# ----------------------------------------------------------------------------
+
+# ----------------------------------------------------------------------------
+# - compiler and linker section                                              -
+# ----------------------------------------------------------------------------
+
+C		= gcc
+CN		= g++
+CC              = g++
+LD              = g++
+LK		= g++
+AR              = ar
+RANLIB		= ranlib
+STDEVFLAGS      =
+STDCCFLAGS      = -Wall -Werror -Wno-overloaded-virtual -fno-builtin
+STACCFLAGS	=
+DYNCCFLAGS	= -fPIC
+PLTCCFLAGS      = 
+DEBUGFLAGS      = -g
+OPTCCFLAGS      = -O2
+PFLCCFLAGS      = -g -pg
+COVCCFLAGS      = -g -fprofile-arcs -ftest-coverage
+CPPCCFLAGS      = -nostdinc -nostdinc++
+CXXCCFLAGS      = 
+STDDEFINES      =
+DBGDEFINES      = -DAFNIX_DEBUG
+OPTDEFINES      =
+PFLDEFINES	= -DAFNIX_DEBUG -DAFNIX_PROFILED
+STDINCLUDE      =
+AFXCPPTYPE	= GNU
+AFXCPPVERS	= 15
+
+# ----------------------------------------------------------------------------
+# - compiler dependant libraries                                             -
+# ----------------------------------------------------------------------------
+
+# adjust for linux platform
+ifeq ($(PLATNAME),linux)
+PLTCCFLAGS      = -MMD -pthread
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# adjust for freebsd platform
+ifeq ($(PLATNAME),freebsd)
+PLTDEFINES	= -MMD -D_REENTRANT
+endif
+
+# adjust for gnu/freebsd platform
+ifeq ($(PLATNAME),gnukbsd)
+PLTDEFINES	= -MMD -D_REENTRANT
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# adjust for gnu platform
+ifeq ($(PLATNAME),gnu)
+PLTDEFINES	= -MMD -D_REENTRANT
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# ----------------------------------------------------------------------------
+# - platform dependant linking flags                                         -
+# ----------------------------------------------------------------------------
+
+# adjust for linux platform
+ifeq ($(PLATNAME),linux)
+ARFLAGS		= rc
+LDFLAGS		= -shared -pthread
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+ifeq ($(OPENMP),yes)
+LDFLAGS        += -fopenmp
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for freebsd platform
+ifeq ($(PLATNAME),freebsd)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for gnu/freebsd platform
+ifeq ($(PLATNAME),gnukbsd)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for gnu platform
+ifeq ($(PLATNAME),gnu)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# ----------------------------------------------------------------------------
+# - platform dependant final executable flags                                -
+# ----------------------------------------------------------------------------
+
+LKFLAGS		=
+
+# adjust linker flags for linux
+ifeq ($(PLATNAME),linux)
+LKFLAGS      	= -pthread
+endif
+
+# adjust linker flags for freebsd
+ifeq ($(PLATNAME),freebsd)
+LKFLAGS      	= -pthread
+endif
+
+# ----------------------------------------------------------------------------
+# - target dependant final executable flags                                  -
+# ----------------------------------------------------------------------------
+
+# adjust extra flag for profiled code
+ifeq ($(CCMODE),profiled)
+LKFLAGS	       += -pg
+endif
+
+# adjust platform flag for static or dynamic
+ifeq ($(LKTYPE),dynamic)
+PLTCCFLAGS     += $(DYNCCFLAGS)
+endif
+ifeq ($(LKTYPE),static)
+PLTCCFLAGS     += $(STACCFLAGS)
+endif
+
+# ----------------------------------------------------------------------------
+# - package dependant flags                                                  -
+# ----------------------------------------------------------------------------
+
+ifeq ($(PKGNAM),deb)
+OPTCCFLAGS     += -g
+endif
+
+ifeq ($(PKGNAM),rpm)
+OPTCCFLAGS     += -g
+endif
diff -urN orig/afnix-src-4.0.0/cnf/mak/afnix-gc16.mak afnix-src-4.0.0/cnf/mak/afnix-gc16.mak
--- orig/afnix-src-4.0.0/cnf/mak/afnix-gc16.mak	1969-12-31 19:00:00.000000000 -0500
+++ afnix-src-4.0.0/cnf/mak/afnix-gc16.mak	2026-07-22 19:54:51.169656289 -0400
@@ -0,0 +1,178 @@
+# ----------------------------------------------------------------------------
+# - afnix-gcc16                                                              -
+# - afnix compiler configuration - gcc configuration                         -
+# ----------------------------------------------------------------------------
+# - This program is  free software;  you can  redistribute it and/or  modify -
+# - it provided that this copyright notice is kept intact.                   -
+# -                                                                          -
+# - This  program  is  distributed in the hope  that it  will be useful, but -
+# - without  any   warranty;  without  even   the   implied    warranty   of -
+# - merchantability  or fitness for a particular purpose. In not event shall -
+# - the copyright holder be  liable for  any direct, indirect, incidental or -
+# - special damages arising in any way out of the use of this software.      -
+# ----------------------------------------------------------------------------
+# - author (c) 1999-2025 amaury   darsch                                     -
+# - author (c) 2011-2013 pino     toscano                      hurd platform -
+# - author (c) 2016-2016 martin   michlmayr                           GCC 05 -
+# - author (c) 2016-2017 amaury   darsch                              GCC 06 -
+# - author (c) 2017-2017 nobuhiro iwamatsu                            GCC 07 -
+# - author (c) 2018-2018 nobuhiro iwamatsu                            GCC 08 -
+# - author (c) 2022-2022 nobuhiro iwamatsu                            GCC 12 -
+# - author (c) 2019-2023 amaury   darsch                     GCC 09/10/11/12 -
+# - author (c) 2022-2023 amaury   darsch                              GCC 13 -
+# ----------------------------------------------------------------------------
+
+# ----------------------------------------------------------------------------
+# - compiler and linker section                                              -
+# ----------------------------------------------------------------------------
+
+C		= gcc
+CN		= g++
+CC              = g++
+LD              = g++
+LK		= g++
+AR              = ar
+RANLIB		= ranlib
+STDEVFLAGS      =
+STDCCFLAGS      = -Wall -Werror -Wno-overloaded-virtual -fno-builtin
+STACCFLAGS	=
+DYNCCFLAGS	= -fPIC
+PLTCCFLAGS      = 
+DEBUGFLAGS      = -g
+OPTCCFLAGS      = -O2
+PFLCCFLAGS      = -g -pg
+COVCCFLAGS      = -g -fprofile-arcs -ftest-coverage
+CPPCCFLAGS      = -nostdinc -nostdinc++
+CXXCCFLAGS      = 
+STDDEFINES      =
+DBGDEFINES      = -DAFNIX_DEBUG
+OPTDEFINES      =
+PFLDEFINES	= -DAFNIX_DEBUG -DAFNIX_PROFILED
+STDINCLUDE      =
+AFXCPPTYPE	= GNU
+AFXCPPVERS	= 16
+
+# ----------------------------------------------------------------------------
+# - compiler dependant libraries                                             -
+# ----------------------------------------------------------------------------
+
+# adjust for linux platform
+ifeq ($(PLATNAME),linux)
+PLTCCFLAGS      = -MMD -pthread
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# adjust for freebsd platform
+ifeq ($(PLATNAME),freebsd)
+PLTDEFINES	= -MMD -D_REENTRANT
+endif
+
+# adjust for gnu/freebsd platform
+ifeq ($(PLATNAME),gnukbsd)
+PLTDEFINES	= -MMD -D_REENTRANT
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# adjust for gnu platform
+ifeq ($(PLATNAME),gnu)
+PLTDEFINES	= -MMD -D_REENTRANT
+ifeq ($(OPENMP),yes)
+PLTCCFLAGS     += -fopenmp
+endif
+endif
+
+# ----------------------------------------------------------------------------
+# - platform dependant linking flags                                         -
+# ----------------------------------------------------------------------------
+
+# adjust for linux platform
+ifeq ($(PLATNAME),linux)
+ARFLAGS		= rc
+LDFLAGS		= -shared -pthread
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+ifeq ($(OPENMP),yes)
+LDFLAGS        += -fopenmp
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for freebsd platform
+ifeq ($(PLATNAME),freebsd)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for gnu/freebsd platform
+ifeq ($(PLATNAME),gnukbsd)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# adjust for gnu platform
+ifeq ($(PLATNAME),gnu)
+ARFLAGS		= rc
+LDFLAGS		= -shared
+ifeq ($(LKMODE),soname)
+LDFLAGS	       += -Wl,-soname,$(SOMAJ)
+endif
+AFXCPPLIBS	=
+endif
+
+# ----------------------------------------------------------------------------
+# - platform dependant final executable flags                                -
+# ----------------------------------------------------------------------------
+
+LKFLAGS		=
+
+# adjust linker flags for linux
+ifeq ($(PLATNAME),linux)
+LKFLAGS      	= -pthread
+endif
+
+# adjust linker flags for freebsd
+ifeq ($(PLATNAME),freebsd)
+LKFLAGS      	= -pthread
+endif
+
+# ----------------------------------------------------------------------------
+# - target dependant final executable flags                                  -
+# ----------------------------------------------------------------------------
+
+# adjust extra flag for profiled code
+ifeq ($(CCMODE),profiled)
+LKFLAGS	       += -pg
+endif
+
+# adjust platform flag for static or dynamic
+ifeq ($(LKTYPE),dynamic)
+PLTCCFLAGS     += $(DYNCCFLAGS)
+endif
+ifeq ($(LKTYPE),static)
+PLTCCFLAGS     += $(STACCFLAGS)
+endif
+
+# ----------------------------------------------------------------------------
+# - package dependant flags                                                  -
+# ----------------------------------------------------------------------------
+
+ifeq ($(PKGNAM),deb)
+OPTCCFLAGS     += -g
+endif
+
+ifeq ($(PKGNAM),rpm)
+OPTCCFLAGS     += -g
+endif
diff -urN orig/afnix-src-4.0.0/src/drv/unx/shl/ctrm.hxx afnix-src-4.0.0/src/drv/unx/shl/ctrm.hxx
--- orig/afnix-src-4.0.0/src/drv/unx/shl/ctrm.hxx	2024-12-24 09:36:45.000000000 -0500
+++ afnix-src-4.0.0/src/drv/unx/shl/ctrm.hxx	2026-07-22 19:54:29.311745678 -0400
@@ -29,6 +29,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_LINUX)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #include <curses.h>
 #include <term.h>
@@ -42,6 +43,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_SOLARIS)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #if (AFNIX_PLATFORM_MAJOR == 5) && (AFNIX_PLATFORM_MINOR < 7)
 #include <curses.h>
@@ -63,6 +65,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_FREEBSD)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #include <curses.h>
 #include <term.h>
@@ -78,6 +81,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_DARWIN)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #include <curses.h>
 #include <term.h>
@@ -94,6 +98,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_GNUKBSD)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #include <curses.h>
 #include <term.h>
@@ -107,6 +112,7 @@
 #if (AFNIX_PLATFORM_PLATID == AFNIX_PLATFORM_GNU)
 #include <stdlib.h>
 #include <unistd.h>
+#include <termios.h>
 #include <sys/ioctl.h>
 #include <curses.h>
 #include <term.h>
diff -urN orig/afnix-src-4.0.0/src/lib/std/shl/Structure.cpp afnix-src-4.0.0/src/lib/std/shl/Structure.cpp
--- orig/afnix-src-4.0.0/src/lib/std/shl/Structure.cpp	2024-12-24 09:36:45.000000000 -0500
+++ afnix-src-4.0.0/src/lib/std/shl/Structure.cpp	2026-07-22 20:09:55.286122250 -0400
@@ -770,7 +770,7 @@
 	// update the structure block
 	p_blok->update (p_desc);
       }
-      if (p_blok->d_ssiz == 0L) {
+      if (p_blok->d_ssiz <= 0L) {
 	unlock ();
 	return false;
       }
