diff -rupN ipbt-20240501.bc876ea/unix/ipbt.c ipbt-20240501.bc876eab/unix/ipbt.c
--- ipbt-20240501.bc876ea/unix/ipbt.c	2024-04-30 21:10:02
+++ ipbt-20240501.bc876eab/unix/ipbt.c	2024-05-03 13:56:59
@@ -17,18 +17,7 @@
 
 #define _XOPEN_SOURCE_EXTENDED
 
-#if CURSES_HAVE_NCURSES_CURSES_H
-#  include <ncursesw/curses.h>
-#elif CURSES_HAVE_NCURSES_CURSES_H
-#  include <ncurses/curses.h>
-#elif CURSES_HAVE_NCURSES_H
-#  include <ncurses.h>
-#elif CURSES_HAVE_CURSES_H
-#  include <curses.h>
-#else
-#  error "SysV or X/Open-compatible Curses header file required"
-#endif
-
+#include <ncurses.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
